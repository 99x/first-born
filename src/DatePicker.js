import React, { Component } from "react";
import { TextInput, Text, StyleSheet, View, Modal, Platform, DatePickerIOS, DatePickerAndroid } from "react-native";
import PropTypes from "prop-types";
import Ionicon from "react-native-vector-icons/Ionicons";

export class DatePicker extends Component {

    constructor(props) {
        super(props);
        this.state = {
            focused: false,
            modalVisible: false,
            defaultDate: props.defaultDate ? props.defaultDate : new Date(),
            chosenDate: !props.placeholder && props.defaultDate ? props.defaultDate : undefined
        }
    }

    render() {
        const { placeholder, color, modalTransparent, animationType, minimumDate, maximumDate, locale, timeZoneOffsetInMinutes } = this.props;

        const underLineColor = !color ? "#000" : color

        if (Platform.OS === "android") {
            return (
                <View style={this.state.focused ? [styles.datePickerAndroid, { borderBottomColor: underLineColor, borderBottomWidth: 2 }] : [styles.datePickerAndroid, { borderBottomColor: underLineColor, borderBottomWidth: 1 }]}>
                    <Text
                        ref={(c) => this._root = c}
                        style={styles.input}
                        onPress={() => this.openAndroidDatePicker()}
                    >
                        {this.state.chosenDate ? this.formatChosenDate(this.state.chosenDate) : !placeholder ? "Select Date" : placeholder}
                    </Text>
                    <Ionicon name="md-calendar" onPress={() => this.openAndroidDatePicker()} style={styles.icon} color={underLineColor} size={20} />
                </View>
            )
        }
        return (
            <View>
                <View style={styles.datePickerIos} >
                    <Text
                        ref={(c) => this._root = c}
                        style={styles.input}
                        onPress={() => this.openIosDatePicker()}
                    >
                        {this.state.chosenDate ? this.formatChosenDate(this.state.chosenDate) : !placeholder ? "Select Date" : placeholder}
                    </Text>
                    <Ionicon onPress={() => this.openIosDatePicker()} name="ios-calendar" style={styles.icon} color={underLineColor} size={20} />
                </View>
                <Modal
                    supportedOrientations={['portrait', 'landscape']}
                    animationType={animationType}
                    transparent={modalTransparent}
                    visible={this.state.modalVisible}
                    onRequestClose={() => { }}
                >
                    <Text
                        onPress={() => this.setState({ modalVisible: false })}
                        style={{ backgroundColor: "#F5FCFF", flex: 1 }}
                    />
                    <DatePickerIOS
                        date={this.state.chosenDate ? this.state.chosenDate : this.state.defaultDate}
                        onDateChange={this.setDate.bind(this)}
                        minimumDate={minimumDate}
                        maximumDate={maximumDate}
                        mode="date"
                        locale={locale}
                        timeZoneOffsetInMinutes={timeZoneOffsetInMinutes}
                    />
                </Modal>
            </View>
        )
    }

    setDate(date) {
        this.setState({ chosenDate: new Date(date) });
        if (this.props.onDateChange) {
            this.props.onDateChange(date);
        }
    }

    openIosDatePicker() {
        this.setState({ modalVisible: true });
    }

    async openAndroidDatePicker() {
        try {
            this.setState({ focused: true });
            const pickedDate = await DatePickerAndroid.open({
                date: this.state.chosenDate ? this.state.chosenDate : this.state.defaultDate,
                minDate: this.props.minimumDate,
                maxDate: this.props.maximumDate,
                mode: this.props.androidMode
            });

            const { action, year, month, day } = pickedDate;

            if (action === DatePickerAndroid.dateSetAction) {
                const selectedDate = new Date(year, month, day);
                this.setDate(selectedDate);
                this.setState({ focused: false });
            }
        } catch ({ code, message }) {
            console.warn('Cannot open date picker', message);
            this.setState({ focused: false });
        }
    }

    formatChosenDate(date) {
        if (this.props.formatChosenDate) {
            return this.props.formatChosenDate(date);
        }
        return [
            date.getDate(),
            date.getMonth() + 1,
            date.getFullYear(),
        ].join('/');
    }

}

DatePicker.propTypes = {
    color: PropTypes.string,
    formatChosenDate: PropTypes.func,
    onDateChange: PropTypes.func,
    defaultDate: PropTypes.object,
    minimumDate: PropTypes.object,
    maximumDate: PropTypes.object,
    locale: PropTypes.string,
    modalTransparent: PropTypes.bool,
    androidMode: PropTypes.string,
    animationType:PropTypes.string,
    ...TextInput.propTypes
}

const styles = StyleSheet.create({
    datePickerAndroid: {
        width: "100%",
        flexDirection: 'row',
        borderTopRightRadius: 3,
        borderTopLeftRadius: 3,
        marginVertical: 10,
        justifyContent: 'center',
        alignItems: 'center',
        height: 50,
        paddingHorizontal: 5,
    },
    datePickerIos: {
        backgroundColor: "white",
        borderRadius: 10,
        paddingHorizontal: 5,
        marginVertical: 10,
        flex: 1,
        borderColor: "#e2e2e2",
        borderWidth: 0.9,
        height: 45,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        height: 50
    },
    icon: {
        paddingRight: 12
    },
    input: {
        flex: 1
    }
})
