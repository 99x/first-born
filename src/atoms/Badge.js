import React, { Component } from "react";
import { View, StyleSheet, Platform, Image } from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class Badge extends Component {
    render() {
        const {
            color,
            children,
            ...otherProps
        } = this.props;

        const badgeColor = color;

        let badgeStyle = [styles.defaultBadge, {
            borderRadius: 15,
            backgroundColor: badgeColor
        }];

        let textStyle = {
            color: commonColors.white
        };

        let iconColor = commonColors.white;;

        const newChildren = React.Children.map(children, child =>
            child && child.type === Text
                ? React.cloneElement(child, {
                    ...child.props,
                    style: { ...child.props.style, ...textStyle },
                    size: "footnote"
                })
                : child && child.type === Icon
                    ? React.cloneElement(child, {
                        ...child.props,
                        color: iconColor,
                        size: 15
                    })
                    : child && child.type === Image
                        ? React.cloneElement(child, {
                            ...child.props,
                            style: {
                                ...child.props.style,
                                width: 15,
                                height: 15
                            }
                        })
                        : null
        );

        return (
            <View
                ref={c => (this._root = c)}
                style={badgeStyle}
                {...otherProps}
            >
                {newChildren}
            </View>
        );
    }
}

Badge.propTypes = {
    color: PropTypes.string,
    ...View.propTypes
};

Badge.defaultProps = {
    color: commonColors.primary,
};

const styles = StyleSheet.create({
    defaultBadge: {
        margin: 5,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        paddingHorizontal: 10,
        paddingVertical: 5
    },
});
