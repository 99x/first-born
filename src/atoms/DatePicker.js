import React, { Component } from "react";
import {
    TextInput,
    StyleSheet,
    View,
    Modal,
    Platform,
    DatePickerIOS,
    DatePickerAndroid,
    TimePickerAndroid
} from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class DatePicker extends Component {
    constructor(props) {
        super(props);
        this.state = {
            focused: false,
            defaultDate: props.defaultDate ? props.defaultDate : new Date(),
            chosenDate:
                !props.placeholder && props.defaultDate
                    ? props.defaultDate
                    : undefined
        };
    }

    render() {
        const {
            placeholder,
            color,
            modalTransparent,
            animationType,
            minimumDate,
            maximumDate,
            locale,
            timeZoneOffsetInMinutes,
            mode
        } = this.props;

        const { focused, chosenDate, defaultDate } = this.state;

        if (Platform.OS === "android") {
            return (
                <View
                    style={
                        focused
                            ? [
                                styles.datePickerAndroid,
                                { borderColor: color, borderWidth: 2 }
                            ]
                            : styles.datePickerAndroid
                    }
                >
                    <Text
                        ref={c => (this._root = c)}
                        style={!chosenDate ? styles.input : styles.inputValue}
                        onPress={() => mode === "date" ? this.openAndroidDatePicker() : this.openAndroidTimePicker()}
                    >
                        {chosenDate
                            ? this.formatChosenDate(chosenDate)
                            : placeholder}
                    </Text>
                    <Icon
                        name="calendar"
                        onPress={() => mode === "date" ? this.openAndroidDatePicker() : this.openAndroidTimePicker()}
                        style={styles.icon}
                        color={focused ? color : commonColors.inputGrey}
                        size={20}
                    />
                </View>
            );
        }
        return (
            <View>
                <View
                    style={
                        focused
                            ? [
                                styles.datePickerIos,
                                { borderColor: color, borderWidth: 2 }
                            ]
                            : styles.datePickerIos
                    }
                >
                    <Text
                        ref={c => (this._root = c)}
                        style={!chosenDate ? styles.input : styles.inputValue}
                        onPress={() => this.openIosDatePicker()}
                    >
                        {chosenDate
                            ? this.formatChosenDate(chosenDate)
                            : placeholder}
                    </Text>
                    <Icon
                        onPress={() => this.openIosDatePicker()}
                        name="calendar"
                        style={styles.icon}
                        color={focused ? color : commonColors.inputGrey}
                        size={20}
                    />
                </View>
                <Modal
                    supportedOrientations={["portrait", "landscape"]}
                    animationType={animationType}
                    transparent={modalTransparent}
                    visible={focused}
                    onRequestClose={() => { }}
                >
                    <Text
                        onPress={() => this.setState({ focused: false })}
                        style={{ flex: 1 }}
                    />
                    <DatePickerIOS
                        date={chosenDate ? chosenDate : defaultDate}
                        onDateChange={this.setDate.bind(this)}
                        minimumDate={minimumDate}
                        maximumDate={maximumDate}
                        mode={mode}
                        locale={locale}
                        timeZoneOffsetInMinutes={timeZoneOffsetInMinutes}
                        style={{ flex: 1, backgroundColor: commonColors.white }}
                    />
                    <Text
                        onPress={() => this.setState({ focused: false })}
                        style={{ flex: 1 }}
                    />
                </Modal>
            </View>
        );
    }

    setDate(date) {
        this.setState({ chosenDate: new Date(date) });
        if (this.props.onDateChange) {
            this.props.onDateChange(date);
        }
    }

    openIosDatePicker() {
        this.setState({ focused: true });
    }

    async openAndroidTimePicker() {
        try {
            this.setState({ focused: true });
            const pickedTime = await TimePickerAndroid.open({
                is24Hour: this.props.is24Hour,
                mode: "default"
            });

            const { action, hour, minute } = pickedTime;

            if (action !== TimePickerAndroid.dismissedAction) {
                const currentDate = new Date();
                const selectedDate = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDay(), hour, minute);
                this.setDate(selectedDate);
            }
            this.setState({ focused: false });
        } catch ({ code, message }) {
            console.warn('Cannot open time picker', message);
            this.setState({ focused: false });
        }
    }

    async openAndroidDatePicker() {
        try {
            this.setState({ focused: true });
            const pickedDate = await DatePickerAndroid.open({
                date: this.state.chosenDate
                    ? this.state.chosenDate
                    : this.state.defaultDate,
                minDate: this.props.minimumDate,
                maxDate: this.props.maximumDate,
                mode: "default"
            });

            const { action, year, month, day } = pickedDate;

            if (action === DatePickerAndroid.dateSetAction) {
                const selectedDate = new Date(year, month, day);
                this.setDate(selectedDate);
            }
            this.setState({ focused: false });
        } catch ({ code, message }) {
            console.warn("Cannot open date picker", message);
            this.setState({ focused: false });
        }
    }

    formatChosenDate(date) {

        if (this.props.formatChosenDate) {
            return this.props.formatChosenDate(date);
        }
        if (this.props.mode === "date") {
            return [date.getDate(), date.getMonth() + 1, date.getFullYear()].join(
                "/"
            );
        } else {
            return [date.getHours(), date.getMinutes()].join(
                ":"
            );
        }
    }
}

DatePicker.propTypes = {
    color: PropTypes.string,
    formatChosenDate: PropTypes.func,
    onDateChange: PropTypes.func,
    defaultDate: PropTypes.object,
    minimumDate: PropTypes.object,
    maximumDate: PropTypes.object,
    locale: PropTypes.string,
    is24Hour: PropTypes.bool,
    modalTransparent: PropTypes.bool,
    animationType: PropTypes.string,
    mode: PropTypes.oneOf(["date", "time"]),
    ...TextInput.propTypes
};

DatePicker.defaultProps = {
    modalTransparent: true,
    color: commonColors.primary,
    animationType: "fade",
    placeholder: "Select Date",
    mode: "date",
    is24Hour: true
};

const styles = StyleSheet.create({
    datePickerAndroid: {
        width: "100%",
        flexDirection: "row",
        marginVertical: 10,
        justifyContent: "center",
        alignItems: "center",
        height: 45,
        paddingHorizontal: 10,
        borderRadius: 10,
        borderColor: commonColors.inputGrey,
        borderWidth: 0.9
    },
    datePickerIos: {
        width: "100%",
        flexDirection: "row",
        borderRadius: 15,
        marginVertical: 10,
        borderColor: commonColors.inputGrey,
        borderWidth: 0.9,
        justifyContent: "center",
        alignItems: "center",
        height: 45,
        paddingLeft: 10
    },
    icon: {
        paddingRight: 15
    },
    input: {
        flex: 1,
        color: commonColors.inputGrey
    },
    inputValue: {
        flex: 1,
        color: commonColors.black
    }
});
