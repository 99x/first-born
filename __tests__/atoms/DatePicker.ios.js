import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { DatePicker } from "../../src/atoms/DatePicker";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "ios";
    return Platform;
});

it("renders regular DatePicker", () => {
    const tree = renderer
        .create(<DatePicker defaultDate={new Date(1554104240475)} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders underlined DatePicker", () => {
    const tree = renderer
        .create(<DatePicker defaultDate={new Date(1554104240475)} underline />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders rounded DatePicker", () => {
    const tree = renderer
        .create(<DatePicker defaultDate={new Date(1554104240475)} rounded />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders unstyled DatePicker", () => {
    const tree = renderer
        .create(<DatePicker defaultDate={new Date(1554104240475)} noStyle />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders custom styled DatePicker", () => {
    const tree = renderer
        .create(
            <DatePicker
                defaultDate={new Date(1554104240475)}
                noStyle
                style={{
                    borderWidth: 1,
                    borderColor: "black",
                    borderRadius: 3
                }}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders colored DatePicker", () => {
    const tree = renderer
        .create(
            <DatePicker color="red" defaultDate={new Date(1554104240475)} />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders DatePicker with placeholder", () => {
    const tree = renderer
        .create(
            <DatePicker
                placeholder="placeholder"
                defaultDate={new Date(1554104240475)}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
