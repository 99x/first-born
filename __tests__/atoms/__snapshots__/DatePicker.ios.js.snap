// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DatePicker with placeholder 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "rgba(33, 33, 33, 0.5)",
          "borderWidth": 0.9,
        },
        Object {
          "borderRadius": 15,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      placeholder
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.5)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTDatePicker
        date={1554104240475}
        mode="date"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 216,
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders colored DatePicker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "rgba(33, 33, 33, 0.5)",
          "borderWidth": 0.9,
        },
        Object {
          "borderRadius": 15,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Date
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.5)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTDatePicker
        date={1554104240475}
        mode="date"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 216,
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders custom styled DatePicker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "black",
          "borderRadius": 3,
          "borderWidth": 1,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Date
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.5)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTDatePicker
        date={1554104240475}
        mode="date"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 216,
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders regular DatePicker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "rgba(33, 33, 33, 0.5)",
          "borderWidth": 0.9,
        },
        Object {
          "borderRadius": 15,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Date
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.5)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTDatePicker
        date={1554104240475}
        mode="date"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 216,
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders rounded DatePicker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "rgba(33, 33, 33, 0.5)",
          "borderWidth": 0.9,
        },
        Object {
          "borderRadius": 15,
          "paddingHorizontal": 10,
        },
        Object {
          "borderRadius": 22.5,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Date
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.5)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTDatePicker
        date={1554104240475}
        mode="date"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 216,
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders underlined DatePicker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderBottomColor": "rgba(33, 33, 33, 0.5)",
          "borderBottomWidth": 0.9,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Date
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.5)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTDatePicker
        date={1554104240475}
        mode="date"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 216,
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders unstyled DatePicker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Date
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.5)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTDatePicker
        date={1554104240475}
        mode="date"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 216,
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;
